commcare-cloud/commcare_cloud/commcare_cloud.py

```diff
@@ -46,6 +47,13 @@ def arg_branch(parser):
     ))
 
 
+def arg_stdout_callback(parser, default='default'):
+    parser.add_argument(
+        '--output', dest='%sstdout_callback' % ENV_ARG_PREFIX, default=default, choices=['actionable', 'minimal'],
+        help=("The callback plugin to use for generating output."
+    ))
```

Comment:
What does setting the `ANSIBLE_STDOUT_CALLBACK` environment variable (or the `stdout_callback` ansible variable do? The [only docs](`http://docs.ansible.com/ansible/latest/intro_configuration.html#stdout-callback`) I could find on it are quite vague. When would you use one or the other?

Reply:
All console output in Ansible is produced by callbacks. Using different callbacks changes the output that is produced.

Here's a list of the standard callbacks:
```
$ ansible-doc -t callback -l
actionable    shows only items that need attention                                                                                                                          
context_demo  demo callback that adds play/task context                                                                                                                     
debug         formated stdout/stderr display                                                                                                                                
default       default Ansible screen output                                                                                                                                 
dense         minimal stdout output                                                                                                                                         
foreman       Sends events to Foreman                                                                                                                                       
full_skip     suppreses tasks if all hosts skipped                                                                                                                          
hipchat       post task events to hipchat                                                                                                                                   
jabber        post task events to a jabber server                                                                                                                           
json          Ansbile screen output as JSON                                                                                                                                 
junit         write playbook output to a JUnit file.                                                                                                                        
log_plays     write playbook output to log file                                                                                                                             
logentries    Sends events to Logentries                                                                                                                                    
logstash      Sends events to Logstash                                                                                                                                      
mail          Sends failure events via email                                                                                                                                
minimal       minimal Ansible screen output                                                                                                                                 
oneline       oneline Ansible screen output                                                                                                                                 
osx_say       oneline Ansible screen output                                                                                                                                 
profile_roles adds timing information to roles                                                                                                                              
profile_tasks adds time information to tasks                                                                                                                                
selective     only print certain tasks                                                                                                                                      
skippy        Ansible screen output that ignores skipped status                                                                                                             
slack         Sends play events to a Slack channel                                                                                                                          
stderr        Splits output, sending failed tasks to stderr                                                                                                                 
syslog_json   sends JSON events to syslog                                                                                                                                   
timer         Adds time to play stats                                                                                                                                       
tree          Save host events to files 
```

You can get details about each one with:
```
$ ansible-doc -t callback {name}
```

Reply:
I've found the default output too verbose so mostly use `actionable` which only shows changes / failures.

Reply:
a5c5b39

Reply:
Awesome, thanks. Had no idea about these!